src/compiler/utilities.ts

```diff
@@ -345,7 +345,7 @@ namespace ts {
     export function getLiteralText(node: LiteralLikeNode, sourceFile: SourceFile) {
         // If we don't need to downlevel and we can reach the original source text using
         // the node's parent reference, then simply get the text as it was originally written.
-        if (!nodeIsSynthesized(node) && node.parent) {
+        if (!nodeIsSynthesized(node) && node.parent && !(isNumericLiteral(node) && node.numericLiteralFlags & TokenFlags.ContainsSeparator)) {

```

Comment:
we should emit them as is for ESNext i would say..

Reply:
Acctually `ES2018` and later.

Reply:
Should we add an `es2018` target now or wait until the ES2018 spec is official?

Reply:
@mhegazy said offline that we should open an issue to add an ES2018 target, since there are some lib additions to make in addition to adding an output mode that preserves these separators.

Reply:
Opened #20342 to track.